frontend/src/pages/solve/[sessionId]/[taskId].tsx

```diff
@@ -25,40 +26,9 @@ const SolveTaskPage = () => {
     return `${scratchAppHostName}/solve/${sessionId}/${taskId}`;
   }, [sessionId, taskId]);
 
+  const [showSessionMenu, setShowSessionMenu] = useState(false);
   const embeddedApp = useRef<EmbeddedAppRef | null>(null);
 
-  const onLoadTask = useCallback(async () => {
-    if (!embeddedApp.current) {
-      return;
-    }
-
-    // TODO: get the task from the API
-    // for now get it from the file system
-
-    const task = await readSingleFileFromDisk();
-
-    await embeddedApp.current.sendRequest({
-      procedure: "loadTask",
-      // typescript does not seem to notice that "arguments" is required but without the Omit<> it does, a bug?
-      arguments: task,
-    });
-  }, []);
-
-  const onSaveTask = useCallback(async () => {
-    if (!embeddedApp.current) {
-      return;
-    }
-
-    const response = await embeddedApp.current.sendRequest({
-      procedure: "getTask",
-    });
-
-    // TODO: Save submission
-
-    // for now, just download the file
-    downloadBlob(response.result, "task.zip");
-  }, []);
-
   const onSubmitSolution = useCallback(async () => {
```

Comment:
This is fine for now, but don't forget that in the future we care about interaction data.
I'm not sure if we (always) want the CRT to be polling the app, we may need to have Scratch pushing stuff (unidirectionally) to the CRT as well. It wouldn't be unreasonable for the solution to be submitted from within Scratch either, considering that different apps may allow submission at different points (e.g. you might not be able to submit a solution unless you've at least tried to run the unit tests).

Reply:
The communication can be initiated by any party, it is just that for now we did not need it the other way around. But it would be trivial to implement :)